File: webgl_engine/open_local.py
"""Open the WebGL engine index.html in the default browser."""
import pathlib
import webbrowser


def main() -> None:
    index = pathlib.Path(__file__).with_name('index.html').resolve()
    webbrowser.open(index.as_uri())


if __name__ == '__main__':
    main()

File: webgl_engine/open_local.js
import { exec } from 'child_process';
import { fileURLToPath } from 'url';
import path from 'path';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const file = path.join(__dirname, 'index.html');

let command;
if (process.platform === 'win32') {
    command = `start "" "${file}"`;
} else if (process.platform === 'darwin') {
    command = `open "${file}"`;
} else {
    command = `xdg-open "${file}"`;
}

exec(command, err => {
    if (err) {
        console.error('Failed to open browser:', err);
    }
});
